contracts/modules/staking/StakingProducts.sol

```diff
@@ -0,0 +1,513 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.9;
+
+import "../../abstract/MasterAwareV2.sol";
+import "../../abstract/Multicall.sol";
+import "../../interfaces/IStakingProducts.sol";
+import "../../libraries/Math.sol";
+import "../../libraries/SafeUintCast.sol";
+import "../../libraries/StakingPoolLibrary.sol";
+
+contract StakingProducts is IStakingProducts, MasterAwareV2, Multicall {
```

Comment:
Naming wise i was expecting contract manages the storage of all products for staking pools.

Then i saw it also has all the pricing logic, so it's more like a toolbox contract now

An observation, i don't have a much better name right now and i know it's already harcoded in a bunch of places

Reply:
Same could be said for the `StakingPool` contract